src/equations/2d/compressible_euler_pot.jl

```diff
@@ -0,0 +1,681 @@
+@doc raw"""
+    CompressibleMoistPotTempEulerEquations2D
```

Comment:
I think we should rename this to `CompressibleEulerMoistEquations2D` (to easily see that it's compressible Euler + some special feature). What do you think @trixi-framework/principal-developers?